### PR TITLE
New apig group resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
In order to manage APIs, environment variables, and custom responses, the APIG (API) group needs support.
The API group is the basic resource for other resources created or updated:
- API creation, publishing and offline
- Variable creation
- Custom response creation

TODO: 
- API, environment variables, domain names and custom response list (for architecture) support.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
reference #1249 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. new api group resource support.
  a. support environment variables management.
2. new related document.
3. new related acc test.
  a. unit test for name and description.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApigGroupV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApigGroupV2_basic -timeout 360m -parallel 4
=== RUN   TestAccApigGroupV2_basic
=== PAUSE TestAccApigGroupV2_basic
=== CONT  TestAccApigGroupV2_basic
--- PASS: TestAccApigGroupV2_basic (447.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      448.041s
```
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApigGroupV2_variables'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApigGroupV2_variables -timeout 360m -parallel 4
=== RUN   TestAccApigGroupV2_variables
=== PAUSE TestAccApigGroupV2_variables
=== CONT  TestAccApigGroupV2_variables
--- PASS: TestAccApigGroupV2_variables (448.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      448.787s
```

### CODE REVIEW

File: docs/resources/apig_api_group.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+* `name` - (Required, String) Specifies the name of the API group.
+  The API group name consists of 3 to 64 characters, starting with a letter.
+  Only letters, digits and underscores (_) are allowed.
+  Chinese characters must be in UTF-8 or Unicode format.
+
+* `description` - (Optional, String) Specifies the description about the API group.
+  The description contain a maximum of 255 characters and the angle brackets (< and >) are not allowed.
+  Chinese characters must be in UTF-8 or Unicode format.
+
+* `environment` - (Required, List) Specifies an array of one or more APIG environments of the associated APIG group.
```
Comment (ShiChangkuo) [inline docs/resources/apig_api_group.md]: this block shoud be optional

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+		// Only letters, digits and underscores (_) are allowed in the name.
+		rName        = fmt.Sprintf("tf_acc_test_%s", acctest.RandString(5))
+		resourceName = "huaweicloud_apig_group.test"
+		group        apigroups.Group
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckEpsID(t)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go]: is `eps_id` necessary?

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	}
+}
+
+func testAccApigGroup_basic(rName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_apig_group" "test" {
+  name        = "%s"
+  instance_id = "76b0e9b706464b82a49813aeffaafcf3"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go]: hard coding is not allowed

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go]: For dedicated instance creation, the `eps_is` is necessary. There is a comment missed, I will make up one.

File: docs/resources/apig_api_group.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/apig_api_group.md]: The type has been updated.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_group_test.go]: The problem has been fixed.

**Pull request merged**